fastlane/lib/fastlane/cli_tools_distributor.rb

```diff
@@ -122,6 +122,28 @@ def take_off
         FastlaneCore::UpdateChecker.show_update_status('fastlane', Fastlane::VERSION)
       end
 
+      # Since loading dotenv should respect additional environments passed using
+      # --env, we must extrat the arguments out of ARGV and process them before
+      # calling into commander. This is required since the ENV must be configured
+      # before running any other commands in order to correclty respect variables
+      # like FASTLANE_HIDE_CHANGELOG and FASTLANE_DISABLE_COLORS
+      def load_dot_env
+        env_cl_param = lambda do
+          index = ARGV.index("--env")
+          return nil if index.nil?
+          ARGV.delete_at(index)
+
+          return nil if ARGV[index].nil?
+          value = ARGV[index]
+          ARGV.delete_at(index)
+
+          value
+        end
```

Comment:
The aim of `env_cl_param` is something like the following:

```ruby
ARGV = ['foo', 'bar', '--env', 'baz', '--verbose']

param = env_cl_param

param # => baz
ARGV # => ['foo', 'bar', '--verbose']
```

I have no idea how to do it nicer though... I think there _must_ be a better way but I can't figure it out

Reply:
Are you popping it off because the CLIs the tools provide define `--env` on their own?

Reply:
@joshdholtz: originally, I **had** to pop it off since not all of the Commander CLIs declared the `global_option("--env")` and would error if I it wasn’t popped from `ARGV`.

Although I’ve since declared the global_option on every CLI, I _could_ leave it in `ARGV`, but the thought was that if a new CLI was added that forgot to declare it, it would cause the “invalid option --env” error despite the fact that that we aren’t expecting the CLI itself to use it.

It’s a bit weird because I pop it, and process it from `CLIToolsDistributor` but have to declare the global_option on each CLI purely for it to show up in `--help`.

Think it will be better to leave it in `ARGV`, or do you have any other ideas?